engine/collection/compliance/core_test.go

```diff
@@ -353,6 +357,34 @@ func (cs *ComplianceCoreSuite) TestOnSubmitVote() {
 	cs.hotstuff.AssertExpectations(cs.T())
 }
 
+// TestOnSubmitTimeout tests that submitting messages.ClusterTimeoutObject adds model.TimeoutObject into
+// TimeoutAggregator.
+func (cs *ComplianceCoreSuite) TestOnSubmitTimeout() {
+	// create a vote
+	originID := unittest.IdentifierFixture()
+	timeout := messages.ClusterTimeoutObject{
+		View:       rand.Uint64(),
+		NewestQC:   helper.MakeQC(),
+		LastViewTC: helper.MakeTC(),
+		SigData:    unittest.SignatureFixture(),
+	}
+
+	cs.timeoutAggregator.On("AddTimeout", &model.TimeoutObject{
+		View:       timeout.View,
+		NewestQC:   timeout.NewestQC,
+		LastViewTC: timeout.LastViewTC,
+		SignerID:   originID,
+		SigData:    timeout.SigData,
+	}).Return()
+
+	// execute the timeout submission
+	err := cs.core.OnTimeoutObject(originID, &timeout)
+	require.NoError(cs.T(), err, "timeout object should pass")
+
+	// check that submit vote was called with correct parameters
+	cs.hotstuff.AssertExpectations(cs.T())
```

Comment:
The timeout is only arriving to the TimeoutAggregator, right? I don't think there are actually any expectations we have for `hotstuff` in this test. 

Since we're using `NewTimeoutAggregator`, we don't need `AssertExpectations` at all, since the constructor registers expectation assertion for when the test ends. 
```suggestion
```